src/com/ms/silverking/cloud/toporing/meta/WeightSpecifications.java

```diff
@@ -72,66 +72,66 @@ public double getWeight(Node node) {
                 return weight;
             }
             */
-            return defaultWeight;
-        }
-    }
-    
-    public List<Double> getWeights(List<Node> nodes) {
-        List<Double>    weights;
-        
-        weights = new ArrayList<>(nodes.size());
-        for (Node node : nodes) {
-            weights.add(getWeight(node));
-        }
-        return weights;
+      return defaultWeight;
     }
-    
-    public WeightSpecifications parse(File file) throws IOException {
-        return parse(new FileInputStream(file));
-    }
-    
-    public WeightSpecifications parse(InputStream inStream) throws IOException {
-        BufferedReader   reader;
-        String           line;
-        
-        reader = new BufferedReader(new InputStreamReader(inStream));
-        do {
-            line = reader.readLine();
-            if (line != null) {
-                line = line.trim();
-                if (line.length() > 0) {
-                    String[]    toks;
-                    
-                    toks = line.split("\\s+");
-                    if (toks.length >= 2) {
-                        setNodeWeightByID(toks[0], Double.parseDouble(toks[1]));
-                    }
-                }
-            }
-        } while (line != null);
-        return this;
-    }
-    
-    public Collection<Map.Entry<String,Double>> getNodeWeights() {
-        return nodeWeights.entrySet();
+  }
+
+  public List<Double> getWeights(List<Node> nodes) {
+    List<Double> weights;
+
+    weights = new ArrayList<>(nodes.size());
+    for (Node node : nodes) {
+      weights.add(getWeight(node));
     }
-    
-    @Override
-    public String toString() {
-        StringBuilder   sb;
-        
-        sb = new StringBuilder();
-        for (Map.Entry<String,Double> entry : nodeWeights.entrySet()) {
-            sb.append(entry.getKey());
-            sb.append('\t');
-            sb.append(entry.getValue());
-            sb.append('\n');
+    return weights;
+  }
+
+  public WeightSpecifications parse(File file) throws IOException {
+    return parse(new FileInputStream(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when WeightSpecifications.parse throws an exception. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).